client/src/templates/Challenges/odin/show.tsx

```diff
@@ -166,33 +176,45 @@ class ShowOdin extends Component<ShowOdinProps, ShowOdinState> {
     }
   }
 
-  handleSubmit(
-    solution: number,
-    openCompletionModal: () => void,
-    assignments: string[]
-  ) {
-    const hasAssignments = assignments.length > 0;
-    const completed = this.state.allAssignmentsCompleted;
-    const isCorrect = solution - 1 === this.state.selectedOption;
-
-    if (isCorrect) {
-      this.setState({
-        isWrongAnswer: false
-      });
-      if (!hasAssignments || completed) openCompletionModal();
-    } else {
-      this.setState({
-        isWrongAnswer: true
-      });
+  handleSubmit = () => {
+    const {
+      data: {
+        challengeNode: {
+          challenge: { questions }
+        }
+      },
+      openCompletionModal
+    } = this.props;
+
+    // subract 1 because the solutions are 1-indexed
+    const mcqSolutions = questions.map(question => question.solution - 1);
+
+    this.setState({
+      submittedMcqAnswers: this.state.selectedMcqOptions,
+      showFeedback: true
+    });
+
+    const allMcqAnswersCorrect = arraysAreEqual(
+      mcqSolutions,
+      this.state.selectedMcqOptions
+    );
```

Comment:
Nit, but I recommend just using `_.isEqual` so we don't have to have yet another tiny util.

Reply:
✅